src/asynk.rs

```diff
@@ -402,6 +402,25 @@ impl From<crate::Message> for Message {
 }
 
 impl Message {
+
+    /// Creates new empty `Message`, without a Client.
+    /// Useful for passing `Message` data, but cannot be used on it's own for associated methods
+    /// as those require `Client` injected into `Message`.
+    pub fn new() -> Message {
+        Default::default()
+    }
+
+    // checks if client is available in Message and if yes, creates owned version for convinient use
+    fn extract_client(&self) -> io::Result<&crate::client::Client> {
```

Comment:
Why not just access directly as Option? 

```rust
if let Some(client) = self.client {}
```

Reply:
We need to handle `None` variant to return error if someone tries to call for example `respond()` on struct with `client: None`.
Match seems more clear than than `if let Some()` with `else`.

Reply:
You can match on Option too yes?

Reply:
Not sure if I follow - that's exactly what happens here: matching on `Option<Client>`

Reply:
You turn it from an Option to a Result, not sure if that incurs malloc etc or not.

Reply:
These checks can be simplified to one liners with `ok_or`.

```rust
let client = self.client.ok_or_else(|| io::Error::new("No client to reply with"))?;
let reply = self.reply.ok_or_else(|| io::Error::new("No reply subject to reply to"))?;
```

Reply:
That's exactly how I was planning to simplify it, without helper func :).